deye_sensors_igen_dtsd422.py

```diff
@@ -0,0 +1,687 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from deye_sensor import (
+    SingleRegisterSensor,
+    ComputedPowerSensor,
+    DoubleRegisterSensor,
+    ComputedSumSensor,
+    SensorRegisterRange,
+)
+
+#
+# Initial IGEN DTSD-422-D3 support.
+#
+# Todo:
+# * Daily Values for Positive and Negative energy are missing, can't find them.
+# * Need to verify data is still read correctly when double reg values exceed one
+#   register
+# * There are some more informational registers, e.g. Manufacturing Date, serial etc. which
+#   we're ignoring for now. Maybe add?
+#
+
+
+class SingleDirectionRegisterSensor(SingleRegisterSensor):
```

Comment:
This class can be renamed to `SignedMagnitudeSingleRegisterSensor` and moved into `deye_sensor.py`
Please add corresponding conversion tests in `deye_sensor_test.py`